rpc_jobs/standard_job_release.yml

```diff
@@ -78,10 +78,203 @@
+    dsl: |
+      library "rpc-gating@${RPC_GATING_BRANCH}"
+      common.globalWraps(){
+        dir("${env.WORKSPACE}/rpc-metadata"){
+          common.clone_with_pr_refs(
+            ".", "git@github.com:${pr_repo}.git", "origin/pr/${pr_number}/merge",
+          )
+          sh """\
+            #!/bin/bash -xe
+            virtualenv --python python3 .venv3
+            set +x; . .venv3/bin/activate; set -x
+            pip install -c constraints.txt -r requirements.txt
+            pushd rpc_component
+                pip install .
+            popd
+          """
+          component = readYaml text: component_text
+          println "=== component CLI standard out ==="
+          println component_text
+
+          String pred_component_text = sh(
+            script: """\
+              #!/bin/bash -xe
+              set +x; . .venv3/bin/activate; set -x
+              src_branch="origin/master"
+              pr_branch="HEAD"
+              component --releases-dir .\
+                 component --component-name ${component["name"]}\
+                 get --version ${component["release"]["version"]["version"]} --pred || true
+            """,
+            returnStdout: true
+          )
+
+          println "=== predecessor component CLI standard out ==="
+          println pred_component_text
+
+          pred_component = readYaml text: pred_component_text
+        }
+
+        ORG="rcbops"
+        REPO=component["name"]
+        VERSION=component["release"]["version"]["version"]
+        PREVIOUS_VERSION=pred_component["release"]["version"]["version"]
+        RELEASE_NOTES_FILE="${WORKSPACE}/artifacts/release_notes"
+        MAINLINE=component["release"]["series"]
+        RC_BRANCH="${MAINLINE}-rc"
+        dir(REPO){
+          common.clone_with_pr_refs(
+            "./",
+            component["repo_url"],
+            "master",
+          )
+          no_rc = sh(
+            script: """\
+              #!/bin/bash -xe
+              git rev-parse --verify ${RC_BRANCH}
+            """,
+            returnStatus: true,
+          )
+        }
+        if (no_rc){
+          release_command = """\
+            python -c 'import sys; print("python rpc-gating/scripts/release.py " + " ".join(sys.argv[1:]))' \
+                --debug \
+                --org "${ORG}" \
+                --repo "${REPO}" \
+                clone \
+                    --ref "${MAINLINE}" \
+                publish_tag \
+                    --ref "${MAINLINE}" \
+                    --version "${VERSION}" \
+                generate_release_notes \
+                    --version "${VERSION}" \
+                    --prev-version "${PREVIOUS_VERSION}" \
+                    --script "optional:gating/generate_release_notes/pre" \
+                    --script "optional:gating/generate_release_notes/run" \
+                    --script "optional:gating/generate_release_notes/post" \
```

Comment:
I know this isn't being executed yet, but what happens if a repo doesn't have release notes and we try to run the above?  Does it cause the script to fail?

Reply:
That's why they're all set to optional, if I read the code correctly it should skip if the script is missing.